[WIP] Document policy.kubernetes.io/engine annotation (#43245)
kubernetes/website: content/en/docs/reference/labels-annotations-taints/_index.md

which is used by Kustomize and similar third-party tools.
For example, Kustomize removes objects with this annotation from its final build output.

### container.apparmor.security.beta.kubernetes.io/* (beta) {#container-apparmor-security-beta-kubernetes-io}

Type: Annotation

Example: `container.apparmor.security.beta.kubernetes.io/my-container: my-custom-profile`

Used on: Pods

This annotation allows you to specify the AppArmor security profile for a container within a
Kubernetes pod. AppArmor is a Linux kernel security module that restricts the capabilities
and access rights of processes within a container, enhancing the security of your workloads. Just look for the [apparmor](https://kubernetes.io/docs/tutorials/security/apparmor/) for more details.

By setting this annotation, you can define a custom AppArmor profile for a container, which dictates the set of rules and restrictions that the containerized process must adhere to. This helps enforce security policies and isolation for your containers.

> 

Author: sorry my bad! I think removing the `container.apparmor.security.beta.kubernetes.io/* (beta) {#container-apparmor-security-beta-kubernetes-io}` make this pr related to the issue.

> 

Reviewer: please always wrap the long lines.

When this annotation is set, the Kubernetes components will "stand-down" and the
`external-provisioner` will act on the objects.

### policy.kubernetes.io/engine

Type: Label

Example: `policy.kubernetes.io/engine: "true"`

Used On: FlowSchema and PriorityLevelConfiguration Objcts

> This label  is used to manage the behavior of the kube-apiserver for specific Kubernetes objects, namely FlowSchema and PriorityLevelConfiguration. When set to "true" on a FlowSchema or PriorityLevelConfiguration object, it signifies that the kube-apiserver will take over the management of the object's specification.

Reviewer 2: "true" -> _true_

Author: Hi Utkarsh, can you please provide some supporting proof of this, according to my research I found this works fine, I am open to make the change if find a solid evidence

Reviewer: Several comments:

- Please avoid using component binary name "kube-apiserver", use "API server" when possible. (xref [documentation style guide](https://kubernetes.io/docs/contribute/style/style-guide/)).
- Wrap the long lines as we did for other paragraphs. This is for ease of change tracking and localization.
- Using quoted "true" is correct, because labels are maps from strings to strings.

Author: @tengqm I am sorry I did not understand which line are you talking about to wrap it up?

Reviewer: Line 736 is not manually wrapped, so it is a loooong line.
Please manually wrap this long lines as we do for other paragraphs.
Git tracks changes on a line-by-line basis.
Long lines are not friendly for future changes or for localizations.

When this annotation is set, the Kubernetes components will "stand-down" and the
`external-provisioner` will act on the objects.

### policy.kubernetes.io/engine

Type: Label

> Example: `policy.kubernetes.io/engine: "true"`

Reviewer 3: `policy.kubernetes.io/engine` is sometimes used as a label in the Policy WG policy report to identify the producer of the report.

Here are some examples:

https://github.com/kubernetes-sigs/wg-policy-prototypes/blob/22764d64b0c3f79d2293b67000b4d9ebca197623/policy-report/samples/sample-falco-policy.yaml#L7

https://github.com/kubernetes-sigs/wg-policy-prototypes/blob/22764d64b0c3f79d2293b67000b4d9ebca197623/policy-report/samples/sample-co.yaml#L6

I am not familiar with any usage in the API server.

Author: I see, in practical terms, API means that certain settings or policies related to traffic flow (in the case of FlowSchema) or priority management (in the case of PriorityLevelConfiguration) are being delegated to the Kubernetes API server for centralized control and enforcement. This can help ensure consistency and centralized management of these configurations across the cluster.

Reviewer 4: This sounds like nonsense I'm afraid. It's English, but it doesn't make sense in the context.

> Used On: FlowSchema and PriorityLevelConfiguration Objcts

Reviewer: ```suggestion
Used on: FlowSchema and PriorityLevelConfiguration
```

Used On: FlowSchema and PriorityLevelConfiguration Objcts

> This label  is used to manage the behavior of the API Server for specific Kubernetes objects, namely "FlowSchema and PriorityLevelConfiguration". 
> 
> When set to "true" on a "FlowSchema or PriorityLevelConfiguration" object, it signifies that the API Server will take over the management of the object's specification.

Reviewer: ```suggestion
This label is used to manage the behavior of the API Server for specific Kubernetes objects,
namely FlowSchema and PriorityLevelConfiguration. 

When the label value is set to "true", it signifies that the API Server will take over
the management of the object's specification.
```
